compiler/src/iree/compiler/Codegen/LLVMGPU/test/ROCDL/config_tile_and_fuse.mlir

```diff
@@ -503,3 +503,60 @@ func.func @scatter_as_root_op(%arg0: tensor<4x?xi64>,
+  %pack = tensor.pack %0 padding_value(%c0_i8 : i8)
+      outer_dims_perm = [0, 1] inner_dims_pos = [0, 1] inner_tiles = [128, 64]
+      into %22 : tensor<1234x567xi8> -> tensor<10x9x128x64xi8>
+  %expanded = tensor.expand_shape %pack [[0], [1], [2, 3, 4], [5, 6, 7]]
+      output_shape [10, 9, 4, 8, 4, 2, 4, 8]
+      : tensor<10x9x128x64xi8> into tensor<10x9x4x8x4x2x4x8xi8>
+  %23 = tensor.empty() : tensor<10x9x8x4x4x4x2x8xi8>
+  %24 = linalg.generic {
+      indexing_maps = [affine_map<(d0, d1, d2, d3, d4, d5, d6, d7) -> (d0, d1, d4, d2, d5, d6, d3, d7)>,
+                       affine_map<(d0, d1, d2, d3, d4, d5, d6, d7) -> (d0, d1, d2, d3, d4, d5, d6, d7)>],
+      iterator_types = ["parallel", "parallel", "parallel", "parallel", "parallel", "parallel", "parallel", "parallel"]}
+      ins(%expanded : tensor<10x9x4x8x4x2x4x8xi8>) outs(%23 : tensor<10x9x8x4x4x4x2x8xi8>) {
+  ^bb0(%in: i8, %out: i8):
+    linalg.yield %in : i8
+  } -> tensor<10x9x8x4x4x4x2x8xi8>
+  return %24 : tensor<10x9x8x4x4x4x2x8xi8>
+}
+
+// CHECK-LABEL: func.func @set_encoding_gpu
+//  CHECK-SAME:   #iree_codegen.translation_info<pipeline = LLVMGPUTileAndFuse workgroup_size = [64, 1, 1] subgroup_size = 64>
+//       CHECK:   linalg.generic {{.*}}lowering_config = #iree_gpu.lowering_config
+//  CHECK-SAME:     thread = [1, 1, 1, 1, 1, 1, 1, 4]
+//  CHECK-SAME:     workgroup = [1, 1, 0, 0, 0, 0, 0, 0]
+
+// -----
+
+func.func @unset_encoding_gpu(%arg0: tensor<10x5x4x8x2x4x16x4xi32>) -> tensor<1234x567xi32> {
+  %c0_i8 = arith.constant 0 : i8
+  %0 = tensor.empty() : tensor<10x5x4x8x4x4x16x2xi32>
+  %transposed = linalg.generic {
+      indexing_maps = [affine_map<(d0, d1, d2, d3, d4, d5, d6, d7) -> (d0, d1, d5, d3, d7, d2, d6, d4)>,
+                       affine_map<(d0, d1, d2, d3, d4, d5, d6, d7) -> (d0, d1, d2, d3, d4, d5, d6, d7)>],
+      iterator_types = ["parallel", "parallel", "parallel", "parallel", "parallel", "parallel", "parallel", "parallel"]}
+      ins(%arg0 : tensor<10x5x4x8x2x4x16x4xi32>) outs(%0 : tensor<10x5x4x8x4x4x16x2xi32>) {
+  ^bb0(%in: i32, %out: i32):
+    linalg.yield %in : i32
+  } -> tensor<10x5x4x8x4x4x16x2xi32>
+  %collapsed = tensor.collapse_shape %transposed [[0], [1], [2, 3, 4], [5, 6, 7]]
+      : tensor<10x5x4x8x4x4x16x2xi32> into tensor<10x5x128x128xi32>
+  %1 = tensor.empty() : tensor<1234x567xi32>
+  %unpack = tensor.unpack %collapsed
+      outer_dims_perm = [0, 1] inner_dims_pos = [0, 1] inner_tiles = [128, 128]
+      into %1 : tensor<10x5x128x128xi32> -> tensor<1234x567xi32>
+  return %unpack : tensor<1234x567xi32>
+}
+
+// CHECK-LABEL: func.func @unset_encoding_gpu
+//  CHECK-SAME:   #iree_codegen.translation_info<pipeline = LLVMGPUTileAndFuse workgroup_size = [64, 1, 1] subgroup_size = 64>
+//       CHECK:   linalg.generic {{.*}}lowering_config = #iree_gpu.lowering_config
+//  CHECK-SAME:     thread = [1, 1, 1, 1, 1, 1, 1, 1]
+//  CHECK-SAME:     workgroup = [1, 1, 0, 0, 0, 0, 0, 0]
```

Comment:
A few more tests to add:
 - Static pack dims but producer is fully dynamic
 - Dynamic pack tile sizes
 - One where the pack tile is almost the entire producer tensor (e.g. <32x32> -> <1x1x32x32>)